src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -437,26 +438,59 @@ public void setConcurrentBuild(boolean b) throws IOException {
     }
 
     @Override public Map<TriggerDescriptor, Trigger<?>> getTriggers() {
-        return triggers.toMap();
+        return getTriggersJobProperty().getTriggersMap();
     }
 
-    public void addTrigger(Trigger trigger) {
-        Trigger old = triggers.get(trigger.getDescriptor());
-        if (old != null) {
-            old.stop();
-            triggers.remove(old);
+    public PipelineTriggersJobProperty getTriggersJobProperty() {
+        PipelineTriggersJobProperty triggerProp = getProperty(PipelineTriggersJobProperty.class);
+
+        if (triggerProp == null) {
+            triggerProp = new PipelineTriggersJobProperty(new ArrayList<Trigger<?>>());
         }
-        triggers.add(trigger);
-        trigger.start(this, true);
+
+        return triggerProp;
     }
 
-    @SuppressWarnings("deprecation")
-    @Override public List<Action> getActions() {
-        List<Action> actions = new ArrayList<Action>(super.getActions());
-        for (Trigger<?> trigger : triggers) {
-            actions.addAll(trigger.getProjectActions());
+    public void setTriggers(List<Trigger<?>> inputTriggers) throws IOException {
+        triggers = null;
+        BulkChange bc = new BulkChange(this);
+        try {
+            PipelineTriggersJobProperty originalProp = getTriggersJobProperty();
+
+            removeProperty(PipelineTriggersJobProperty.class);
+
+            PipelineTriggersJobProperty triggerProp = new PipelineTriggersJobProperty(inputTriggers);
+
+            addProperty(triggerProp);
+            bc.commit();
+
+            originalProp.stopTriggers();
+
+            triggerProp.startTriggers(Items.currentlyUpdatingByXml());
```

Comment:
`submit` is already (effectively) calling `startTriggers` via `PipelineTriggersJobProperty.reconfigure`.